apps/desktop/src/lib/backend/updater.ts

```diff
@@ -1,124 +1,161 @@
-import { listen } from './ipc';
+import { Tauri } from './tauri';
 import { showToast } from '$lib/notifications/toasts';
-import { getVersion } from '@tauri-apps/api/app';
 import { relaunch } from '@tauri-apps/api/process';
 import {
-	checkUpdate,
 	installUpdate,
-	onUpdaterEvent,
 	type UpdateResult,
 	type UpdateManifest,
 	type UpdateStatus
 } from '@tauri-apps/api/updater';
 import posthog from 'posthog-js';
-import { derived, writable, type Readable } from 'svelte/store';
-
-// TODO: Investigate why 'DOWNLOADED' is not in the type provided by Tauri.
-export type Update = {
-	version?: string;
-	status?: UpdateStatus | 'DOWNLOADED';
-	body?: string;
-	manual: boolean;
-};
-
+import { derived, readable, writable } from 'svelte/store';
+
+type Status = UpdateStatus | 'DOWNLOADED';
+const TIMEOUT_SECONDS = 30;
+
+/**
+ * Note that the Tauri API `checkUpdate` hangs indefinitely in dev mode, build
+ * a nightly if you want to test the updater manually.
+ *
+ * export TAURI_PRIVATE_KEY=doesnot
+ * export TAURI_KEY_PASSWORD=matter
+ * ./scripts/release.sh --channel nightly --version "0.5.678"
+ */
```

Comment:
@PavelLaptev you should be able to build a nightly with a specific version using this command.